File: sublib/src/main/java/com/enhtmv/sublib/common/util/CookieUtil.java
package com.enhtmv.sublib.common.util;

import com.enhtmv.sublib.common.SubReport;
import com.enhtmv.sublib.common.http.SubResponse;

import java.net.HttpCookie;
import java.util.ArrayList;
import java.util.List;

public class CookieUtil {


    public static String parse(String str) {


        List<String> cookies = new ArrayList<>();

        for (HttpCookie httpCookie : HttpCookie.parse(str)) {

            String cookie = String.format("%s=%s", httpCookie.getName(), httpCookie.getValue());
            cookies.add(cookie);
        }

        return StringUtil.join(";", cookies);

    }


    public static String parse(SubResponse sub) {

        List<String> cookies = new ArrayList<>();

        for (String str : sub.response().headers("Set-Cookie")) {
            cookies.add(parse(str));
        }

        return StringUtil.join(";", cookies);

    }


}

File: sublib/src/main/java/com/mos/lib/common/util/CookieUtil.kt
package com.mos.lib.common.util

import com.mos.lib.common.http.SubResponse

import java.net.HttpCookie
import java.util.ArrayList

object CookieUtil {


    fun parse(str: String): String {


        val cookies = ArrayList<String>()

        for (httpCookie in HttpCookie.parse(str)) {

            val cookie = String.format("%s=%s", httpCookie.name, httpCookie.value)
            cookies.add(cookie)
        }

        return StringUtil.join(";", cookies)

    }


    fun parse(sub: SubResponse): String {

        val cookies = ArrayList<String>()

        for (str in sub.response().headers("Set-Cookie")) {
            cookies.add(parse(str))
        }

        return StringUtil.join(";", cookies)

    }


}
